src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/PrisonerContactRegistryClient.kt

```diff
@@ -26,4 +36,61 @@ class PrisonerContactRegistryClient(
       .bodyToMono<List<PrisonerContactDto>>()
       .block(apiTimeout)
   }
+
+  fun getBannedRestrictionDateRange(prisonerId: String, visitors: List<Long>, prisonDateRange: DateRange): DateRange {
+    val uri = "/prisoners/$prisonerId/approved/social/contacts/restrictions/banned/dateRange"
+
+    return webClient.get()
+      .uri(uri) {
+        getBannedRestrictionDateRangeParams(visitors, prisonDateRange, it)
+      }
+      .retrieve()
+      .bodyToMono<DateRange>()
+      .onErrorResume {
+          e ->
+        if (!isNotFoundError(e)) {
+          LOG.error("getBannedRestrictionDateRage Failed get request $uri")
+        } else {
+          LOG.error("getBannedRestrictionDateRage NOT FOUND get request $uri")
+        }
```

Comment:
This might have to change to cater for results from PCR, if the BANNED restriction has no expire date the PCR might throw a not found exception. This then should be picked up and a NO range exception should be thrown.